### PR TITLE
Default implementation of under/over speed checks

### PR DESCRIPTION
**Pull request by spilkey-cisco**
#### Description
Provide default implementation of fan under and over speed threshold checks, providing backwards compatibility for vendors that only implement `get_speed_tolerance`

#### Motivation and Context
Fan under/over speed checks should be vendor customizable, since a tolerance based off the pwm/percentage fan speed can easily give false failures, especially for low fan speeds.

#### How Has This Been Tested?
```
root@sonic:/home/cisco# echo 10000 > /opt/cisco/etc/fantray0.fan0.rpm
root@sonic:/home/cisco# grep thermalctld /var/log/syslog
<snip>
May 19 05:09:47.763970 sonic WARNING pmon#thermalctld: Fan high speed warning: fantray0.fan0 current speed=91, target speed=20
May 19 05:09:51.129298 sonic INFO pmon#supervisord 2023-05-19 05:09:51,128 INFO success: thermalctld entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
May 19 05:10:01.347935 sonic INFO pmon#supervisord: thermalctld WARNING:cisco.pacific.thermal.thermal_zone:level minor: fantray0.fan0: pwm 20; motor out of tolerance @ rpm 10000; maximum rpm 2950
root@sonic:/home/cisco# echo 2400 > /opt/cisco/etc/fantray0.fan0.rpm
root@sonic:/home/cisco# grep thermalctld /var/log/syslog
<snip>
May 19 05:09:47.763970 sonic WARNING pmon#thermalctld: Fan high speed warning: fantray0.fan0 current speed=91, target speed=20
May 19 05:09:51.129298 sonic INFO pmon#supervisord 2023-05-19 05:09:51,128 INFO success: thermalctld entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
May 19 05:10:01.347935 sonic INFO pmon#supervisord: thermalctld WARNING:cisco.pacific.thermal.thermal_zone:level minor: fantray0.fan0: pwm 20; motor out of tolerance @ rpm 10000; maximum rpm 2950
May 19 05:10:47.023156 sonic NOTICE pmon#thermalctld: Fan high speed warning cleared: fantray0.fan0 speed is back to normal
```

### CODE REVIEW

**Review by amulyan7 — APPROVED** [review]
LGTM

File: sonic_platform_base/fan_base.py [hunk 1]
```diff
@@ -66,6 +66,44 @@ def get_speed_tolerance(self):
...
         raise NotImplementedError
 
+    def is_under_speed(self):
+        """
+        Calculates if the fan speed is under the tolerated low speed threshold
+
+        Returns:
+            A boolean, True if fan speed is under the low threshold, False if not
+        """
+        speed = self.get_speed()
```
Comment (bmridul) [inline sonic_platform_base/fan_base.py:80]: This file seems to have stub implementations. Do we need to provide a reference implementation (which may not work for some vendor..)

Comment (spilkey-cisco) [reply sonic_platform_base/fan_base.py:80]: This default implementation is provided for backwards compatibility. Any vendors who were happy with the way tolerance calculations worked do not need to change their code; they continue to just have `get_speed_tolerance` implemented, and these default implementations of `is_under/over_speed` do the previously expected tolerance calculation.

**Comment by spilkey-cisco** [comment]
Closing to open new PR to merge into sonic-net/sonic-platform-common

**Comment by spilkey-cisco** [comment]
https://github.com/sonic-net/sonic-platform-common/pull/382

**Pull request closed without merging**